lib/tables.c

```diff
@@ -103,7 +103,7 @@ expand_column(void **column, size_t new_max_rows, size_t element_size)
     void *tmp;
 
     tmp = realloc((void **) *column, new_max_rows * element_size);
-    if (tmp == NULL) {
+    if (tmp == NULL && (new_max_rows * element_size > 0)) {
```

Comment:
Happily, `malloc` and `realloc` etcetera behave sensibly when dealing with chunks of memory of size zero, as we get with `spatial_dimension` of zero.

Reply:
Unhappily though this isn't portable: https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations

In reality we have to catch zero lengh allocs and explicitly avoid them. I'll often do something like MSP_MAX(size, 1) just to make sure we allocate at least one byte.

Reply:
=(